### ISSUE TITLE
BusinessPack issue with DatePicker control

### ISSUE THREAD

**Issue by tunapps** [riganti/dotvvm#832]
```
<bp:DatePicker SelectedDate="{value: SelectedDate}" DataContext="{value: MyDatePicker}" 
MinDate="{value: MinimumDate}" MaxDate="{value: MaximumDate}" FormatString="d" 
FirstDayOfWeek="Monday" Restrictions="{value: DateRestrictions}" >  </bp:DatePicker>

<p>{{value: MyDatePicker.SelectedDate}}</p>
```
I've got following problem on rendering: 

```
Uncaught (in promise) ReferenceError: SelectedDate is not defined
    at eval (eval at DotvvmEvaluator.evaluateOnViewModel (dotvvm--internal:1), <anonymous>:1:61)
    at DotvvmEvaluator.evaluateOnViewModel (dotvvm--internal:1)
    at u.showValidationErrorsFromServer (dotvvm--internal:1)
    at DotvvmEventHandler.handler (dotvvm--internal:1)
    at e.trigger (dotvvm--internal:1)
    at dotvvm--internal:1
```
In addition, there is also a problem with event "Changed" that the command doesn't not get called (just only static command). Wherever I use the control, I can't get call preRender event in viewmodel.

Asp.net core project
BusinessPack: 2.4.0.4.-preview04-final
Dotvvm for VS: 2.4.172.0
Dotvvm.aspnetcore 2.4.0.1

**Comment by tomasherceg** [comment]
The stack trace shows there is some problem with validation. Is the page using validation? 
By any chance, are you adding something in the ModelState manually, or are you implementing `IValidatableObject`? It looks like when a validation error is generated using a C# code, the property path is not correct.

**Comment by tunapps** [comment]
I guess I found a problem because you led me to validation. When I create DateRangeRestriction, I must not divide the interval of range restriction into two intervals. If I joined these two intervals into one range, everything is OK. Thank you very much for your help.

**Comment by tomasherceg** [comment]
This looks like a bug in the control - I'll report that to the Business Pack team so it can get fixed.

**Comment by tunapps** [comment]
I discovered another problem with this control and DateRangeRestriction. When I set the restriction on interval 10 Mai 2020 (10.05.2020 0:00:00) -11 Mai 2020 (12.05.2020 0:00:00) and I would like to choose 12 Mai 2020, command in "Changed" event also doesn't fired. Please fix them all :)

**Comment by mrnustik** [comment]
Hi @tunapps,
I am from the Business Pack team, we have now released the 2.4.0.4.-preview06-final in which this behavior should be fixed. We will release a non-preview version shortly afterwards.
